kong/router.lua

```diff
@@ -789,23 +783,78 @@ end
 
 
 local function sort_sources(r1, _)
-  for _, source in ipairs(r1.sources) do
-    if source.ip and source.port then
+  for i = 1, r1.sources[0] do
+    if r1.sources[i].ip and r1.sources[i].port then
       return true
     end
   end
 end
 
 
 local function sort_destinations(r1, _)
-  for _, destination in ipairs(r1.destinations) do
-    if destination.ip and destination.port then
+  for i = 1, r1.destinations[0] do
+    if r1.destinations[i].ip and r1.destinations[i].port then
```

Comment:
I can check. In general I try to avoid too excessive localizing of these. Like couple of times is fine and probably more performant than localizing for just two uses. But I recheck this.

Reply:
i guess localizing just `r1.destinations` gets the biggest speed vs stack use ratio.  but of course, testing is better than guesses

Reply:
@javierguerragiraldez, I'll do that change.

Reply:
This was changed by localizing `r1.destinations`